### ISSUE TITLE
vmware-iso builder hangs on ssh if it reboots during provisioning phase.

### ISSUE THREAD

**Issue by ameyakarkhanis** [mitchellh/packer#4512]
I am using vmware-iso builder to create a debian 8.6 image with some custom software installed as part of provisioning phase. During the provisioning phase I need to perform reboot before continuing on next  script for installation. I am using shell provisioner and for restart I am using recommended process of first killing sshd and then calling reboot as follows:
```
killall sshd;
shutdown -r now;
sleep 120;
```

When the Vm reboots packer detects that previous provisioning script exited without exit status or signal.  When VM comes up it gets a new IP address but packer still tries to SSH over old IP address.

On digging further I noticed packer initially figures out the IP address by looking up the mac address of VM in /var/db/vmware/vmnet-dhcpd-vmnet8.leases file. 

Before the reboot contents of this file were: 
```
lease 192.168.136.241 {
	starts 1 2017/02/03 11:22:03;
	ends 1 2018/02/06 11:37:03;
	hardware ethernet 00:0c:29:35:88:cb;
	client-hostname "debian";
}
```
After VM reboots contents of the file were:
```
lease 192.168.136.241 {
	starts 1 2017/02/03 11:22:03;
	ends 1 2018/02/06 11:37:03;
	abandoned;
}
lease 192.168.136.242 {
	starts 1 2017/02/03 11:25:52;
	ends 1 2018/02/06 11:40:52;
	hardware ethernet 00:0c:29:35:88:cb;
	client-hostname "debian";
}
```

But during this reboot packer does not watch "vmnet-dhcpd-vmnet8.leases" file to lookup ip address using mac address. 

Few details on my environment:
- Packer version: 0.12.2
- Host platform: OSX El Capitan v10.11.6
- Partial debug log output showing the behavior of using same IP after reboot:
https://gist.github.com/ameyakarkhanis/7e340b94d43c590e1125fc8f9916e74b
- Sample template.json file:
https://gist.github.com/ameyakarkhanis/5395f875f3b33ffdbf506f7e9e573385

**Comment by mwhooker** [comment]
I believe this is a dupe of #3585 which has been fixed. If you still run into this after building from master, please let me know and I'll re-open

**Comment by ameyakarkhanis** [comment]
I will give this a try but I was running this using Packer Version : 0.12.2 which was released on Jan 20, 2017. Based on the date on bug you mentioned as duplicate that version should already have that fix.

**Comment by mwhooker** [comment]
it was fixed in #4532 which was only merged in 18 days ago, which was after 0.12.2. The next release will happen later this week
